src/main/java/dev/ecattez/shahmat/board/Typed.java

```diff
@@ -0,0 +1,7 @@
+package dev.ecattez.shahmat.board;
+
+public interface Typed {
+
+    boolean isOfType(PieceType type);
```

Comment:
`sameTypeAs` ?

Reply:
It could work with a `Typed` parameter but it is useless